packages/controllers/src/plugins/PluginController.ts

```diff
@@ -53,38 +53,56 @@ type RemoveAllPermissionsFunction = (pluginIds: string[]) => void;
 type CloseAllConnectionsFunction = (domain: string) => void;
 type RequestPermissionsFunction = (
   domain: string,
-  requestedPermissions: IRequestedPermissions,
-) => IOcapLdCapability[];
+  requestedPermissions: RequestedPluginPermissions,
+) => Promise<IOcapLdCapability[]>;
 type HasPermissionFunction = (
   domain: string,
   permissionName: string,
 ) => boolean;
 type GetPermissionsFunction = (domain: string) => IOcapLdCapability[];
+type TerminateWorkerOf = (pluginName: string) => void;
+type Command = (
+  workerId: string,
+  message: JsonRpcRequest<unknown>,
+) => Promise<unknown>;
+type TerminateAll = () => void;
+type CreatePluginWorker = (metadate: PluginWorkerMetadata) => Promise<string>;
+type StartPlugin = (
+  workerId: string,
+  pluginData: PluginData,
+) => Promise<unknown>;
 
-interface StoredPlugins {
-  [pluginId: string]: Plugin;
-}
+type StoredPlugins = Record<string, Plugin>;
```

Comment:
So that it's immediately apparent what the keys "are", we should either use object syntax to define this type or create a `string` type named `PluginId`. I think I prefer the latter.